fdbserver/workloads/BitFlipping.actor.cpp

```diff
@@ -28,85 +28,202 @@
+			auto itr = workersMap.find(workerAddress);
+			if (itr != workersMap.end())
+				self->doBitFlips(itr->second, self->percentBitFlips);
+		}
+		return Void();
+	}
+	// For fetching chaosMetrics to ensure chaos events are happening
+	// This is borrowed code from Status.actor.cpp
+	struct WorkerEvents : std::map<NetworkAddress, TraceEventFields> {};
+
+	ACTOR static Future<Optional<std::pair<WorkerEvents, std::set<std::string>>>> latestEventOnWorkers(
+	    std::vector<WorkerDetails> workers,
+	    std::string eventName) {
+		try {
+			state vector<Future<ErrorOr<TraceEventFields>>> eventTraces;
+			for (int c = 0; c < workers.size(); c++) {
+				EventLogRequest req =
+				    eventName.size() > 0 ? EventLogRequest(Standalone<StringRef>(eventName)) : EventLogRequest();
+				eventTraces.push_back(errorOr(timeoutError(workers[c].interf.eventLogRequest.getReply(req), 2.0)));
+			}
+
+			wait(waitForAll(eventTraces));
+
+			std::set<std::string> failed;
+			WorkerEvents results;
+
+			for (int i = 0; i < eventTraces.size(); i++) {
+				const ErrorOr<TraceEventFields>& v = eventTraces[i].get();
+				if (v.isError()) {
+					failed.insert(workers[i].interf.address().toString());
+					results[workers[i].interf.address()] = TraceEventFields();
+				} else {
+					results[workers[i].interf.address()] = v.get();
+				}
+			}
+
+			std::pair<WorkerEvents, std::set<std::string>> val;
+			val.first = results;
+			val.second = failed;
+
+			return val;
+		} catch (Error& e) {
+			ASSERT(e.code() ==
+			       error_code_actor_cancelled); // All errors should be filtering through the errorOr actor above
+			throw;
+		}
+	}
+
+	// Fetches chaosMetrics and verifies that chaos events are happening for enabled workers
+	ACTOR static Future<Void> chaosGetStatus(BitFlippingWorkload* self) {
+		std::vector<WorkerDetails> workers =
+		    wait(self->dbInfo->get().clusterInterface.getWorkers.getReply(GetWorkersRequest{}));
+
+		Future<Optional<std::pair<WorkerEvents, std::set<std::string>>>> latestEventsFuture;
+		latestEventsFuture = latestEventOnWorkers(workers, "ChaosMetrics");
+		state Optional<std::pair<WorkerEvents, std::set<std::string>>> workerEvents = wait(latestEventsFuture);
+
+		state WorkerEvents cMetrics = workerEvents.present() ? workerEvents.get().first : WorkerEvents();
+
+		// Now verify that all chosen workers for chaos events have non-zero chaosMetrics
+		for (auto& workerAddress : self->chosenWorkers) {
+			auto chaosMetrics = cMetrics.find(workerAddress);
+			if (chaosMetrics != cMetrics.end()) {
+				int bitFlips = chaosMetrics->second.getInt("BitFlips");
+
+				// we expect bitFlips to be non-zero for chosenWorkers
+				if (bitFlips == 0) {
+					TraceEvent(SevError, "ChaosGetStatus")
+					    .detail("OnEndpoint", workerAddress.toString())
+					    .detail("BitFlips", bitFlips);
+				}
+			}
+		}
+
+		return Void();
+	}
+
+	// Periodically fetches chaosMetrics to ensure that chaas events are taking place
```

Comment:
This could result in failures due to unlucky timing.  I think it's safe to just check this once, in the `check()` phase of the workload.  It's true this could miss worker instances that are no longer alive but I think it is sufficient to prove that the most recent generation of workers with the chosen addresses did respond correctly.